cocos/particle/billboard.ts

```diff
@@ -113,6 +113,54 @@ export class Billboard extends Component {
         }
     }
 
+    @serializable
+    private _techIndex = 0;
+
+    @tooltip('i18n:billboard.technique')
+    public get technique (): number {
+        return this._techIndex;
+    }
+
+    public set technique (val: number) {
+        // clamp technique index
+        val = Math.floor(val);
+        if (val < 0) {
+            val = 0;
+        }
+        const techs = this._material?.effectAsset?.techniques;
+        if (techs && val >= techs.length) {
+            val = techs.length - 1;
```

Comment:
Is it a problem if techs.length is 0?

Reply:
Yeah, have never thought of. Theoretically it is possible.
I will change the code.